### PR TITLE
Deterministic unit test for periodic gaussians

### PR DESCRIPTION
**Pull request by mcbennet**
Periodic gaussians -- Carbon diamond off gamma.
The test verifies that all MOs match the ones
produced from pyscf along a path that contains
regions both inside and outside the prim cell.

This PR addresses issue #2012

### CODE REVIEW

**Comment by prckent** [comment]
Please can you add a short comment to the files about how they were generated, could be updated etc.

**Comment by prckent** [comment]
OK to test

**Comment by prckent** [comment]
Thanks for the added comments. LGTM. Someone non-local is needed to review+approve.

File: src/QMCWaveFunctions/tests/test_pyscf_complex_MO.cpp [hunk 1]
```diff
@@ -0,0 +1,905 @@
...
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2018 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+/*  Author: Chandler Bennett
```
Comment (markdewing) [inline src/QMCWaveFunctions/tests/test_pyscf_complex_MO.cpp]: This attribution can be moved to the "File created by" and/or "File developed by" header above.

**Pull request merged**